// asaas-sdk/lib/index.js
//const config = require('./config');
const constants = require("./constants");
const customers = require("./customers");
const payments = require("./payments");
const subscriptions = require("./subscriptions");

/**
 * Classe principal para a integração com a API do serviço Asaas.com.br
 * @method config
 */
class asaasSDK {
  constructor() {
    this.SANDBOX = constants.SANDBOX;
    this.PRODUCTION = constants.PRODUCTION;

    this._config = {
      environment: constants.SANDBOX,
      apiKey: "",
      version: "v3",
      debug: false,
    };

    this.customers = new customers(this.config);
    this.payments = new payments(this.config);
    this.subscriptions = new subscriptions(this.config);
  }

  /**
   * Função para atribuir ou obter as configurações do componente.
   * @param {Object} conf Objeto JSON com os parâmetros de configuração.
   * @returns Objeto JSON {environment: string, apiKey: string, version: string, debug: boolean}
   */
  config = (conf) => {
    this._config = Object.assign(this._config, conf);

    return this._config;
  };
}

module.exports = new asaasSDK();
